src/PrintReceipt.ts

```diff
@@ -1,12 +1,120 @@
-import {loadAllItems, loadPromotions} from './Dependencies'
+import { loadAllItems, loadPromotions } from "./Dependencies";
+import { Item, Promotion, Quantity, ReceiptItem, Tag } from "./Interface";
 
 export function printReceipt(tags: string[]): string {
-  return `***<store earning no money>Receipt ***
-Name：Sprite，Quantity：5 bottles，Unit：3.00(yuan)，Subtotal：12.00(yuan)
-Name：Litchi，Quantity：2.5 pounds，Unit：15.00(yuan)，Subtotal：37.50(yuan)
-Name：Instant Noodles，Quantity：3 bags，Unit：4.50(yuan)，Subtotal：9.00(yuan)
-----------------------
-Total：58.50(yuan)
-Discounted prices：7.50(yuan)
-**********************`
+  return renderReceipt(generateReceiptItems(parseTags(tags)));
+
+  //   return `***<store earning no money>Receipt ***
+  // Name：Sprite，Quantity：5 bottles，Unit：3.00(yuan)，Subtotal：12.00(yuan)
+  // Name：Litchi，Quantity：2.5 pounds，Unit：15.00(yuan)，Subtotal：37.50(yuan)
+  // Name：Instant Noodles，Quantity：3 bags，Unit：4.50(yuan)，Subtotal：9.00(yuan)
+  // ----------------------
+  // Total：58.50(yuan)
+  // Discounted prices：7.50(yuan)
+  // **********************`;
+}
+
+function parseTags(tags: string[]): Tag[] {
+  const parseTags: Tag[] = [];
+  for (let tag of tags) {
+    const parseTag = tag.split("-");
+    if (parseTag.length === 1) {
+      parseTags.push({ barcode: parseTag[0], quntity: 1 });
+    } else {
+      parseTags.push({ barcode: parseTag[0], quntity: Number(parseTag[1]) });
+    }
+  }
+  return parseTags;
+}
+
+function generateReceiptItems(tags: Tag[]): ReceiptItem[] {
+  const items: Item[] = loadAllItems();
+  const promotions: Promotion[] = loadPromotions();
+
+  const receiptItems: ReceiptItem[] = [];
+
+  for (let tag of tags) {
+    let find = 0;
+    let existInReceiptItems = 0;
+    let name: string = "";
+    let unit: string = "";
+    let unitPrice: number = 0;
+    let subtotal: number = 0;
+    let discountedPrice: number = 0;
+
+    for (let item of items) {
+      if (item.barcode === tag.barcode) {
+        name = item.name;
+        unitPrice = item.price;
+        unit = item.unit;
+        find = 1;
+      }
+    }
+    if (find === 0) throw "Don't exist";
+
+    for (let receiptItem of receiptItems) {
+      if (name === receiptItem.name) {
+        receiptItem.quantity.value += 1;
+        existInReceiptItems = 1;
+
+        break;
+      }
+    }
+
+    if (existInReceiptItems === 0) {
+      receiptItems.push({
```

Comment:
For the push(object), you can create a empty items object and set it's value, this can make sure each item's value is correct.